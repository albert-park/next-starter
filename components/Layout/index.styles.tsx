import styled from 'styled-components';

export const StyledHero = styled.div`
  width: 100%;
  color: #333
`;

export const StyledTitle = styled.h1`
    margin: 0;
    width: 100%;
    padding-top: 80px;
    line-height: 1.15;
    font-size: 48px;
    text-align: center;
`;

export const StyledDescripttion = styled.p`
  text-align: center;
`;
