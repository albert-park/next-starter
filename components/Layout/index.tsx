import React, { FC } from 'react';
import Head from 'next/head';
import Nav from '../Nav';
import { StyledHero, StyledTitle, StyledDescripttion } from './index.styles';

interface LayoutProps {
  title: string,
  layoutStyle?: object
}

const Layout: FC<LayoutProps> = ({ children, title }) => (
  <StyledHero>
    <Nav />
    <StyledTitle>
        Welcome to Next.js Starter
      {' '}
      {title}
      {' '}
Page!
    </StyledTitle>
    <StyledDescripttion>
        To get started, edit
      {' '}
      <code>pages/index.js</code>
      {' '}
        and save to reload.
    </StyledDescripttion>
    {children}
  </StyledHero>
);

export default Layout;
