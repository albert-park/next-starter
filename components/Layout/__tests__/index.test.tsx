import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Layout from '../index';

const mockLayoutProps = {
  title: 'Test title',
};

const mockHeadline = `Welcome to Next.js Starter ${mockLayoutProps.title} Page!`;

describe('Layout Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Layout {...mockLayoutProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders headline with title prop', () => {
    const { getByText } = render(<Layout {...mockLayoutProps} />);
    expect(getByText(mockHeadline)).toHaveTextContent(mockHeadline);
    expect(getByText(mockHeadline)).toBeVisible();
  });
});
