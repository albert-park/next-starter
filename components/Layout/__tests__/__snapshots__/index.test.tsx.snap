// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="indexstyles__StyledHero-sc-17i3g0v-0 vlZQG"
  >
    <nav
      class="indexstyles__StyledNav-mi8ir5-0 gfWmhc"
    >
      <ul
        class="indexstyles__StyledUl-mi8ir5-1 hfekUk"
      >
        <li
          class="indexstyles__StyledLi-mi8ir5-2 gaXxcC"
        >
          <a
            class="indexstyles__StyledAnchor-mi8ir5-3 hUFsIF"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="indexstyles__StyledLi-mi8ir5-2 gaXxcC"
        >
          <a
            class="indexstyles__StyledAnchor-mi8ir5-3 hUFsIF"
            href="https://zeit.co/now"
          >
            ZEIT
          </a>
        </li>
        <li
          class="indexstyles__StyledLi-mi8ir5-2 gaXxcC"
        >
          <a
            class="indexstyles__StyledAnchor-mi8ir5-3 hUFsIF"
            href="https://github.com/zeit/next.js"
          >
            GitHub
          </a>
        </li>
      </ul>
    </nav>
    <h1
      class="indexstyles__StyledTitle-sc-17i3g0v-1 kpdsfj"
    >
      Welcome to Next.js Starter Test title Page!
    </h1>
    <p
      class="indexstyles__StyledDescripttion-sc-17i3g0v-2 hZrcbT"
    >
      To get started, edit 
      <code>
        pages/index.js
      </code>
       and save to reload.
    </p>
  </div>
</DocumentFragment>
`;
